File: problem/Construct_Min_Bitwise_Array_I/minBitwiseArray.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
const minBitwiseArray = nums => {
    const res = [];
    for (const n of nums) {
        if (n & 1)
            res.push(n & ~(((n + 1) & ~n) >> 1));
        else
            res.push(-1);
    }
    return res;
};

File: problem/Construct_Min_Bitwise_Array_I/minBitwiseArray.c
int *minBitwiseArray(int *nums, int numsSize, int *returnSize)
{
    int *res = (int *)malloc(sizeof(int) * numsSize);
    *returnSize = numsSize;

    for (int i = 0; i < numsSize; i++)
    {
        int n = nums[i];
        if (n & 1)
        {
            unsigned int z = ((unsigned int)(n + 1) & ~n) >> 1;
            res[i] = n & ~z;
        }
        else
        {
            res[i] = -1;
        }
    }

    return res;
}